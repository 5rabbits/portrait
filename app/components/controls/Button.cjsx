React = require('react')


@RoundButton = React.createClass
  render: ->
    return (
      <a href="#" className="btn btn-outline-primary btn-sm block-rounded">
        <i className="fa fa-fw fa-btn fa-{@props.icon}"></i>
        { @props.label }
      </a>
    )

@Button = React.createClass
  render: ->
    return (
      <a href="#" className="btn btn-primary">{ @props.label }</a>
    )


module.exports = @Button