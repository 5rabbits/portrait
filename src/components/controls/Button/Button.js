import React from 'react'
import PropTypes from 'prop-types'

const Button = props =>
  <a href={props.href} className="btn btn-primary">
    {props.label}
  </a>

Button.propTypes = {
  href: PropTypes.string,
  label: PropTypes.string,
}

Button.defaultProps = {
  href: null,
  label: null,
}

export default Button
