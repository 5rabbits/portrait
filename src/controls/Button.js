import React from 'react'

const Button = () =>
  <a href="#" className="btn btn-primary">
    {this.props.label}
  </a>

export default Button
