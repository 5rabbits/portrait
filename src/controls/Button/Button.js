import React, { PropTypes } from 'react'

const Button = props =>
  <a href={props.href} className="btn btn-primary">
    {props.label}
  </a>

Button.propTypes = {
  href: PropTypes.string,
  label: PropTypes.string,
}

export default Button
