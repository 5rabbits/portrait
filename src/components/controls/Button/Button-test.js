import { React, mount, toJSON } from 'test/helper'
import Button from 'controls/Button'

describe('Button', () => {
  it('should render with minimum props', () => {
    const component = mount(<Button />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
