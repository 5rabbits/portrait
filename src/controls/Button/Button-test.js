import { React, expect, mount } from 'test/helper'
import Button from 'controls/Button'

describe('Button', () => {
  it('should render with minimum props', () => {
    const component = mount(<Button />)

    expect(component).to.be.ok
  })
})
