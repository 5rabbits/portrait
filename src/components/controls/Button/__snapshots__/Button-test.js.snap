// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with minimum props 1`] = `
<Button>
  <a
    className="btn btn-primary"
  />
</Button>
`;
